#165593726 Implement getByEmail endpoint

okabamac opened the pull request:
## What does this PR do?
Implement the getByEmail endpoint

Remove email from token

Change validation error status code

Move validation middleware to routes

## Description of Task to be completed?

have the following endpoint working
GET /api/v1/users/:email/accounts

# How should this be manually tested?
clone the repo git clone git@github.com:okabamac/banka.git

on your terminal run cd journal

run git checkout ft-165582997

run npm install

run npm run dev

make a GET request to the route

Any background context you want to provide?

NA

## What are the relevant pivotal tracker stories?

[#165593726](https://www.pivotaltracker.com/story/show/165582997)
[#165601668](https://www.pivotaltracker.com/story/show/165601668)

okabamac committed:
ft(getByEmail) Implement getByEmail Endpoint

- Implement getByEmail
- Write unit tests
- Change validation

[Delivers #165582997]

okabamac committed:
Added .vscode to gitignore

Mrshinezee commented on .gitignore:
```diff
@@ -1,2 +1,3 @@
-.env
\ No newline at end of file
+.env
+.vscode
```
Add new line

okabamac replied:
OK

Mrshinezee commented on src/controllers/accountController.js:
```diff
@@ -77,17 +77,18 @@ class AccountControl {
 
     if (validCreateAccount.statusCode === 422) return;
```
Why are you checking status code.

okabamac replied:
The middleware I am using returns unhandled promise if I dont

Mrshinezee commented on src/controllers/userController.js:
```diff
@@ -39,6 +39,31 @@ class UserControl {
+      email,
+    } = req.params;
+
+    if (email == req.decoded.email) {
```
Remove email from token

okabamac replied:
These are the things I will be working on the feedback

Mrshinezee commented:
ft(getByEmail)-165582997-user-can-view-by-email - Check the documentation on how it should look.

okabamac committed:
ch(feedback) Implemented feedback

Remove email from token
Move validations to routes
Change status code of 422 to 400
Write more test
Finished the getByEmail Endpoint
[Finishes #165601668]
